### ISSUE TITLE
Switch config file format to json

### ISSUE THREAD

**Issue by dfornika** [BCCDC-PHL/illumina-uploader#63]
JSON is a more flexible format for storing config info. It supports arrays and associative data, with arbitrary nesting.

**Comment by dfornika** [comment]
For this iteration I'd suggest making essentially a one-to-one conversion of the config file into json format. We'll probably want to re-format things a bit later but doing it this way should minimize the changes necessary in the codebase.

Current config template:

```ini
[LOCAL]
inputdirs = /path/to/local/input/directory/1,/path/to/local/input/directory/2
sequencer = miseq
folderregexmiseq = \d{6}_[A-Z0-9]{6}_\d{4}_\d{9}-[A-Z0-9]{5}$
folderregexnextseq = \d{6}_[A-Z0-9]{7}_\d{1,}_[A-Z0-9]{9}$
watchfilepath =  CompletedJobInfo.xml
logfile = logfile.log
sleeptime = 5

[SERVER]
host = hostname.com
pemfile = /path/to/hostname.pem
loginid = firstname.lastname
outputdirs = /path/to/remote/output/directory/1,/path/to/remote/output/directory/2

[EMAIL]
enabled = false
emailurl = https://emailprovider.com
emailtoken = secrettoken
mailto = recipient
mailsubject = [illumina-uploader] {status} miseq data transfer to sabin
mailbody = Run directory {folderToUpload} {status} transferring to Sabin at {timeOfMail}

[COMMANDS]
rsynccommand = rsync -artvh -p -e "{sshformat} -i {pem}" {chmod} {inDir}/{inFile} {login}@{host}:{outDir}
scpcommand = scp -i {pem} {inDir}/{filename} {login}@{host}:{outDir}/{filename}
sshnixcommand = ssh
sshwincommand = /usr/bin/ssh
chmodcommand = --chmod=Du=rwx,Dg=rx,Fu=rw,Fg=r,o=

[DB]
location = local.db
backupfolder = backups
foldertable = folderinfo

[SQL]
createtable = CREATE TABLE {} (folder text, status text, querylastrun text);
checkfolderpresence = SELECT * FROM {} WHERE folder='{}';
insertfolder = INSERT INTO {} VALUES ('{}','{}','{}');
getfolderstoupload = SELECT {} FROM {} WHERE status<>'{}';
markfileindb = UPDATE {} SET status='{}' WHERE folder='{}';
```

proposed json config template:

```json
{
  "LOCAL": {
    "inputdirs": [
      "/path/to/local/input/directory/1",
      "/path/to/local/input/directory/2"
    ],
    "sequencer": "miseq",
    "folderregexmiseq": "\\d{6}_[A-Z0-9]{6}_\\d{4}_\\d{9}-[A-Z0-9]{5}$",
    "folderregexnextseq": "\\d{6}_[A-Z0-9]{7}_\\d{1,}_[A-Z0-9]{9}$",
    "watchfilepath": "CompletedJobInfo.xml",
    "logfile": "logfile.log",
    "sleeptime": 5
  },
  "SERVER": {
    "host": "hostname.com",
    "pemfile": "/path/to/hostname.pem",
    "loginid": "firstname.lastname",
    "outputdirs": [
      "/path/to/remote/output/directory/1",
      "/path/to/remote/output/directory/2"
    ]
  },
  "EMAIL": {
    "enabled": false,
    "emailurl": "https://emailprovider.com",
    "emailtoken": "secrettoken",
    "mailto": "recipient",
    "mailsubject": "[illumina-uploader] {status} miseq data transfer to sabin",
    "mailbody": "Run directory {folderToUpload} {status} transferring to Sabin at {timeOfMail}"
  },
  "COMMANDS": {
    "rsynccommand": "rsync -artvh -p -e \"{sshformat} -i {pem}\" {chmod} {inDir}/{inFile} {login}@{host}:{outDir}",
    "scpcommand": "scp -i {pem} {inDir}/{filename} {login}@{host}:{outDir}/{filename}",
    "sshnixcommand": "ssh",
    "sshwincommand": "/usr/bin/ssh",
    "chmodcommand": "--chmod=Du=rwx,Dg=rx,Fu=rw,Fg=r,o="
  },
  "DB": {
    "location": "local.db",
    "backupfolder": "backups",
    "foldertable": "folderinfo"
  },
  "SQL": {
    "createtable": "CREATE TABLE {} (folder text, status text, querylastrun text);",
    "checkfolderpresence": "SELECT * FROM {} WHERE folder='{}';",
    "insertfolder": "INSERT INTO {} VALUES ('{}','{}','{}');",
    "getfolderstoupload": "SELECT {} FROM {} WHERE status<>'{}';",
    "markfileindb": "UPDATE {} SET status='{}' WHERE folder='{}';"
  }
}
```

**Comment by dfornika** [comment]
Fixed in #66
